glossary.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Glossary 
+description: A collection of technical terminology commonly used in the Astro documentation and community.
+i18nReady: false
+---
+
+This glossary describes common terminology used in the Astro documentation and the Astro community.
+
+{ /* // Uncomment this section when we have more terms to add to the glossary.
+It is separated into Astro-specific terms and general web development terms.
+## Astro-Specific Terms 
+
+Some future guidelines for this:
+- Astro-Specific: Content Collections, ViewTransitions, Astro Component
+- General: Editor, LSP, Front-End Library, Framework, Metaframework, Framework Component
+*/}
+
+### Component
+
+Astro components are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime. You can spot an Astro component by its file extension: `.astro`. Read more about components in the [Astro Components](https://docs.astro.build/en/core-concepts/astro-components/) page.
+
+### Islands
+
+**Islands** is a frontend architecture that results in better frontend performance by helping you avoid monolithic JavaScript patterns and stripping all non-essential JavaScript from the page automatically. The technique that this architectural pattern builds on is also known as partial or selective hydration. In Astro, an `island` refers to any interactive UI component on the page. Read more about islands in the [Astro Islands](https://docs.astro.build/en/concepts/islands/) page.
```

Comment:
```suggestion
An [Astro Island](https://docs.astro.build/en/concepts/islands/) is an [interactive UI framework component](https://docs.astro.build/en/core-concepts/framework-components/) on the page.

**Islands** is a frontend architecture that results in better frontend performance by helping you avoid monolithic JavaScript patterns and stripping all non-essential JavaScript from the page automatically. The technique that this architectural pattern builds on is also known as partial or selective hydration. In Astro,  Read more about islands in the  page.
```
So, I didn't edit here, but I will point out that I think it's way more helpful for a glossary entry to not bury the lede. 😅  If people wanted to read docs, they'd.... read docs?

FOR ALL OF YOU:

I think part of making these entries helpful is deciding what your *purpose* is. I would argue that (mostly) repeating docs is not a terribly useful thing to have, nor for us to spend time on, when there are so many other things to do.

So, I think before all of you start adding more and more entries, really think about what this glossary is going to ACHIEVE. How should it be different from what I've already written, and edited, in docs itself? How are you going to make it *useful* and worthwhile? WHO is going to read this over docs, and for what goal, and how can you cater it to THAT audience to serve that purpose?

Reply:
While the edit was included, the overarching question remains.

I would think search serves the purpose of "I am not familiar with this term". Is there a case where search is inadequate and a glossary page is necessary?